Domain/Serialization/Serializer.cs

```diff
@@ -161,15 +162,16 @@ public static T FromJsonTo<T>(this string json)
             string body,
             dynamic uniqueEventId = null,
             JsonSerializerSettings serializerSettings = null,
+            DeserializeEvent deserializationFunc = null,
             string etag = null)
         {
             var deserializerKey = aggregateName + ":" + eventName;
- 
-            serializerSettings = serializerSettings ?? Settings;
+
+            deserializationFunc = deserializationFunc ?? BuildDeserializationFunc(serializerSettings);
```

Comment:
Performing this coalesce operation in the constructor of the repository will benefit performance, since it will be done once per repository instance instead of once per deserialized event.

Reply:
DeserializeEvent is a public method so other callers will require BuildDeserializer, right?.  Other callers are expecting that if serializerSettings is null, then Settings is used with JsonConvert.DeserializeObject.  DeserializeEvent is called by EventExtensions.ToDomainEvent, which itself is public and also has it's own logic for building serialization settings (by cloning and swapping out the ContractResolver).  ToDomainEvent is called by the repository.  I don't see a good way to push this back into the Repository, but I'm sure I'm just missing it.

Reply:
I would have to expose Serializer.BuildDeserializer and also expose EventExtensions.serializerSettings  or else duplicate the code, right?

Reply:
I'd force the caller to pass something. Those methods can be internal as I believe Its.Domain.Sql can call internals of Its.Domain.  Would that work?

Duplication is preferable to changing the public interface since it's easier to refactor later.

Reply:
I can work with that.   will follow-up with a commit to fix